backend/src/services/course-category/course-category.service.ts

```diff
@@ -0,0 +1,22 @@
+import { CourseCategoryGetResponseDto } from '~/common/types/types';
+import { courseCategory as courseCategoryRep } from '~/data/repositories/repositories';
+
+type Constructor = {
+  courseCategoryRepository: typeof courseCategoryRep;
+};
+
+class CourseCategory {
+  #courseCategoryRepository: typeof courseCategoryRep;
+
+  constructor({ courseCategoryRepository }: Constructor) {
+    this.#courseCategoryRepository = courseCategoryRepository;
+  }
+
+  async getByName(name: string): Promise<CourseCategoryGetResponseDto | null> {
+    const courseCategory = await this.#courseCategoryRepository.getByName(name);
+
+    return courseCategory ?? null;
```

Comment:
```suggestion
  async getByName(name: string): Promise<CourseCategoryGetResponseDto | null> {
    return this.#courseCategoryRepository.getByName(name);

```

`this.#courseCategoryRepository.getByName(name)` already returns `courseCategory ?? null`, no need to check it again